samples/rust/src/walletcore_iface.rs

```diff
@@ -0,0 +1,190 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+// Interfacing to wallet-core: interface methods, type utilites, wrappers
+
+use libc::c_char;
+use std::ffi::CString;
+use std::ffi::CStr;
+use hex::ToHex;
+
+// signatures
+extern "C" {
+    fn TWStringCreateWithUTF8Bytes(twstring: *const c_char) -> *const u8;
+    fn TWStringDelete(twstring: *const u8);
+    fn TWStringUTF8Bytes(twstring: *const u8) -> *const c_char;
+
+    fn TWDataCreateWithBytes(bytes: *const u8, size: usize) -> *const u8;
+    fn TWDataDelete(twstring: *const u8);
+    fn TWDataSize(data: *const u8) -> usize;
+    fn TWDataBytes(data: *const u8) -> *const u8;
+
+    fn TWPrivateKeyData(private_key: *const u8) -> *const u8;
+    fn TWPrivateKeyGetPublicKeySecp256k1(private_key: *const u8, compressed: bool) -> *const u8;
+    fn TWPrivateKeyDelete(twstring: *const u8);
+
+    fn TWPublicKeyData(private_key: *const u8) -> *const u8;
+    fn TWPublicKeyDelete(twstring: *const u8);
+
+    fn TWHDWalletCreateWithMnemonic(mnemonic: *const u8, passphrase: *const u8) -> *const u8;
+    fn TWHDWalletDelete(twstring: *const u8);
+    fn TWHDWalletGetAddressForCoin(wallet: *const u8, coin: u32) -> *const u8;
+    fn TWHDWalletGetKeyForCoin(wallet: *const u8, coin: u32) -> *const u8;
+
+    fn TWAnySignerSign(input: *const u8, coin: u32) -> *const u8;
+
+    fn TWMnemonicIsValid(mnemonic: *const u8) -> bool;
+}
+
+// type utilities
+pub struct TWString {
```

Comment:
Move `TWString` and `TWData` code to separate files, we can reuse them later (the reset can be generated)

Reply:
Good insight; I separated it into interface code that could be generated and some extra traits, for TWString and TWData.

Reply:
****